common/src/main/java/gov/cms/ab2d/common/service/PdpClientServiceImpl.java

```diff
@@ -33,6 +37,14 @@
 
     private final Mapping mapping;
 
+    @Override
+    public List<Contract> getAllEnabledContracts() {
+        return pdpClientRepository.findAllByEnabledTrue().stream()
+                .filter(client -> client.getContract().getAttestedOn() != null)
+                .filter(client -> client.getRoles().stream().anyMatch(role -> role.getName().equals(SPONSOR_ROLE)))
```

Comment:
Stream within a stream? Maybe put the internal stream into another method like:
client -> hasRole(client, SPONSOR_ROLE)

Reply:
Unwrapped it see if you approve.